File: 06_salary_increase_by_category/python/main.py
if __name__ == '__main__':
    print("\n--- PROGRAMA DE AUMENTO DE SALARIO POR CATEGORIAS ---\n")

    category = input("Ingresa la categoria a la que perteneces (A, B, C) > ")

    if category.lower() != 'a' and category.lower() != 'b' and category.lower() != 'c':
        print(f'La categoria "{category}" ingresada no existe')
    else:
        salary = float(input("Ingresa tu salario actual > "))

        if category.lower() == 'a':
            increment_percentage = 0.15
        elif category.lower() == 'b':
            increment_percentage = 0.1
        else:
            increment_percentage = 0.07

        increment = salary * increment_percentage
        new_salary = salary + increment

        print(f"\nTu aumento será de: $ {new_salary}")
        print(f"Tu nuevo salario será de: $ {new_salary}")

File: 06_salary_increase_by_category/rust/src/main.rs
use std::io;
use std::io::Write;

fn main() {
    let mut input = String::new();

    println!("\n--- PROGRAMA DE AUMENTO DE SALARIO POR CATEGORIAS ---\n");

    print!("Ingresa la categoria a la que perteneces (A, B, C) > ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut input).unwrap();
    let category = input.trim().to_string();

    if category.to_lowercase() != "a" && category.to_lowercase() != "b" && category.to_lowercase() != "c" {
        println!("La categoria \"{category}\" ingresada no existe")
    } else {
        print!("Ingresa tu salario actual > ");
        io::stdout().flush().unwrap();
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let salary = input.trim().parse::<f32>().expect("El salario debe ser un número");

        let increment_percentage = if category.to_lowercase() == "a" {
            0.15_f32
        } else if category.to_lowercase() == "b" {
            0.1
        } else {
            0.07
        };

        let increment = salary * increment_percentage;
        let new_salary = salary + increment;

        println!("\nTu aumento será de: $ {increment}");
        println!("Tu nuevo salario será de: $ {new_salary}");
    }
}
